lib/http.js

```diff
@@ -8,11 +8,28 @@ const bedrock = require('bedrock');
 const brPassport = require('bedrock-passport');
 const {profiles, profileAgents} = require('bedrock-profile');
 const {validate} = require('bedrock-validation');
-const {
-  ensureAuthenticated
-} = brPassport;
+const {ensureAuthenticated} = brPassport;
 
 const {config, util: {BedrockError}} = bedrock;
+const cfg = config['profile-http'];
+
+let privateKmsBaseUrl;
+let publicKmsBaseUrl;
+
+bedrock.events.on('bedrock.init', () => {
+  if(cfg.kms.private.port === 443) {
+    privateKmsBaseUrl = `https://${cfg.kms.private.host}/kms`;
+  } else {
+    privateKmsBaseUrl =
+      `https://${cfg.kms.private.host}:${cfg.kms.private.port}/kms`;
+  }
+  if(cfg.kms.public.port === 443) {
+    publicKmsBaseUrl = `https://${cfg.kms.public.host}/kms`;
+  } else {
+    publicKmsBaseUrl =
+      `https://${cfg.kms.public.host}:${cfg.kms.public.port}/kms`;
+  }
```

Comment:
We should just make these computed vars in the config so we don't need to do this init handler.

Reply:
@dlongley making them computed does not solve the problem does it?  If they are not set yet, the logic here will fail right?

Reply:
Our general rule is to make config things that are composed of other config things use computed configs. It also allows others to access these variables and the code is in one place.

Reply:
by config things do you mean 'bedrock.config' things, because privateKmsBaseUrl  and publicKmsBaseUrl are not config things, but are derived from config things.

Making cfg.kms.foo a computed config does not solve for having to compute these strings in an `init` handler does it?

Reply:
You still have to wait to access them after the configuration system has initialized. However, you can remove this `init` handler and just destructure them from the config in the router configuration handler below.